proseco/guide.py

```diff
@@ -30,6 +30,7 @@ def get_guide_catalog(obsid=0, **kwargs):
     :param date: date of acquisition (any DateTime-compatible format)
     :param dither: dither size 2-element tuple: (dither_y, dither_z) (float, arcsec)
     :param n_guide: number of guide stars to attempt to get
+    :param known_fids: selected fids (used for guide star exclusion)
```

Comment:
You don't need to invent something new here.  The `fids` property will already handle it, so the only change to have the `GuideTable` know about fids is to call `get_guide_catalog(stars=stars, fids=aca.fids, **kwargs)`.  Look at `acq.py` for a model of how that works.

There should not even be an issue in pickling / garbage collection because there is no circular reference (guide => fid => guide).  Also, my understanding is that pickling will only store an object once and otherwise just store refs to that object.

Reply:
I saw all kinds of stuff related to fids in `acq.py` but still haven't figured out what to do to just get it to be an allowed keyword arg for GuideTable.

Reply:
If I try to just supply it and I get

```
-> 101     aca.guides = get_guide_catalog(stars=stars, fids=aca.fids, **kwargs)
    102 
    103     # Make a merged starcheck-like catalog.  Catch any errors at this point to avoid

/proj/sot/ska/jeanproj/git/proseco/proseco/guide.py in get_guide_catalog(obsid, **kwargs)
     39 
     40     guides = GuideTable()
---> 41     guides.set_attrs_from_kwargs(obsid=obsid, **kwargs)
     42     guides.set_stars()
     43 

/proj/sot/ska/jeanproj/git/proseco/proseco/core.py in set_attrs_from_kwargs(self, **kwargs)
    222                 setattr(self, name, val)
    223             else:
--> 224                 raise ValueError(f'unexpected keyword argument "{name}"')
    225 
    226         # If an explicit obsid is not provided to all getting parameters via mica

ValueError: unexpected keyword argument "fids"

```

It thought it would be set as an allowed keyword arg on 'guide' by MetaAttribute.__set_name__ but that didn't seem to get called and I couldn't figure it out.

Reply:
You need this, which you can put into this PR:
```
diff --git a/proseco/guide.py b/proseco/guide.py
index f1e027f..198d614 100644
--- a/proseco/guide.py
+++ b/proseco/guide.py
@@ -71,6 +71,8 @@ class GuideTable(ACACatalogTable):
     # (e.g. `obs19387/guide.yaml`).
     name = 'guide'
 
+    allowed_kwargs = ACACatalogTable.allowed_kwargs | set(['fids'])
+
     # Required attributes
     required_attrs = ('att', 't_ccd_guide', 'date', 'dither_guide')
```